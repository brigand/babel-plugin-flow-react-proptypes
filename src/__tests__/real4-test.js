const babel = require('babel-core');
const content = `
// @flow
import type { AdPartner, Job, JobAlert, SeoText } from './JLCommon';
import type { DirectApplyDraft, DirectApplyFile } from './reducer/DirectApply';
import type { SearchQuery } from './joblift-query-parser';

// eslint-disable-next-line
export type { SearchQuery };

export type jpopunderPostResult = {
    items: Job[],
    adPartners: AdPartner[],
};

export type searchPostResult = {
    items: Job[],
    seoText?: SeoText[],
    query: SearchQuery,
    index: boolean,
    total: number,
    pages: number,
    resultId: string,
    facets?: { [key: string]: { key: number, amount: number, translation: string }[] },
};

// api/jobalert/me/
export type jobalertMeGetResult = {
    description: string,
    jobalerts: JobAlert[],
};

// api/joblocation-reverse-lookup
export type joblocationReverseLookupGetResult = {
    display_name: string,
};

// api/joblocation-suggestion
export type joblocationSuggestionGetResult = {
    suggestions: Array<{ value: string }>,
};

// api/jobsearch-suggestion
export type jobsearchSuggestionGetResult = {
    suggestions: Array<{ value: string, numJobs?: number }>,
};

// api/jobsearch-suggestion/popular
export type freqSearchSuggestionGetResult = {
    suggestions: Array<{ value: string, numJobs: number }>,
};

// api/charts
export type statisticalDataGetResult = {
    keys: string[],
    data: Object,
};

// api/joboffer
export type jobofferGetResult = Job;

// api/popunder
export type popunderPostResponse = {
    showPopunder: boolean,
    url: string,
    itemCount: number,
    partner: string,
    customer: string,
    customerId: number,
    suppressTransactionId?: boolean,
    ch: string,
};

// auth/json/login/local
export type loginPostResult = {
    displayName?: string,
    firstName: string,
    initialCreated: boolean,
    isAnonymous?: boolean,
    lastName: string,
    localAccounts: Array<{
        email: string,
    }>,
    socialAccounts: Object[],
    uniqueEmails: string[],
    uuid: string,
};

// api/direct-apply/offer/<jobofferid>/draft
export type directApplyDraftGet = DirectApplyDraft;

// api/direct-apply/files/recent
export type directApplyRecentFilesGet = DirectApplyFile[];
`;

it('real4', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
