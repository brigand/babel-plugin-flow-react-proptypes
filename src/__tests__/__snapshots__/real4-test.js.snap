// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`real4 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
    value: true
});
exports.bpfrpt_proptype_directApplyRecentFilesGet = exports.bpfrpt_proptype_directApplyDraftGet = exports.bpfrpt_proptype_loginPostResult = exports.bpfrpt_proptype_popunderPostResponse = exports.bpfrpt_proptype_jobofferGetResult = exports.bpfrpt_proptype_statisticalDataGetResult = exports.bpfrpt_proptype_freqSearchSuggestionGetResult = exports.bpfrpt_proptype_jobsearchSuggestionGetResult = exports.bpfrpt_proptype_joblocationSuggestionGetResult = exports.bpfrpt_proptype_joblocationReverseLookupGetResult = exports.bpfrpt_proptype_jobalertMeGetResult = exports.bpfrpt_proptype_searchPostResult = exports.bpfrpt_proptype_jpopunderPostResult = undefined;

var _JLCommon = require('JLCommon');

var _DirectApply = require('reducer/DirectApply');

var _jobliftQueryParser = require('joblift-query-parser');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var bpfrpt_proptype_jpopunderPostResult = {
    items: _propTypes2.default.arrayOf(function () {
        return (typeof _JLCommon.bpfrpt_proptype_Job === 'function' ? _JLCommon.bpfrpt_proptype_Job : _propTypes2.default.shape(_JLCommon.bpfrpt_proptype_Job)).apply(this, arguments);
    }).isRequired,
    adPartners: _propTypes2.default.arrayOf(function () {
        return (typeof _JLCommon.bpfrpt_proptype_AdPartner === 'function' ? _JLCommon.bpfrpt_proptype_AdPartner : _propTypes2.default.shape(_JLCommon.bpfrpt_proptype_AdPartner)).apply(this, arguments);
    }).isRequired
};
var bpfrpt_proptype_searchPostResult = {
    items: _propTypes2.default.arrayOf(function () {
        return (typeof _JLCommon.bpfrpt_proptype_Job === 'function' ? _JLCommon.bpfrpt_proptype_Job : _propTypes2.default.shape(_JLCommon.bpfrpt_proptype_Job)).apply(this, arguments);
    }).isRequired,
    seoText: _propTypes2.default.arrayOf(function () {
        return (typeof _JLCommon.bpfrpt_proptype_SeoText === 'function' ? _JLCommon.bpfrpt_proptype_SeoText : _propTypes2.default.shape(_JLCommon.bpfrpt_proptype_SeoText)).apply(this, arguments);
    }),
    query: function query() {
        return (typeof _jobliftQueryParser.bpfrpt_proptype_SearchQuery === 'function' ? _jobliftQueryParser.bpfrpt_proptype_SearchQuery.isRequired ? _jobliftQueryParser.bpfrpt_proptype_SearchQuery.isRequired : _jobliftQueryParser.bpfrpt_proptype_SearchQuery : _propTypes2.default.shape(_jobliftQueryParser.bpfrpt_proptype_SearchQuery).isRequired).apply(this, arguments);
    },
    index: _propTypes2.default.bool.isRequired,
    total: _propTypes2.default.number.isRequired,
    pages: _propTypes2.default.number.isRequired,
    resultId: _propTypes2.default.string.isRequired,
    facets: _propTypes2.default.shape({})
};

// eslint-disable-next-line

var bpfrpt_proptype_jobalertMeGetResult = {
    description: _propTypes2.default.string.isRequired,
    jobalerts: _propTypes2.default.arrayOf(function () {
        return (typeof _JLCommon.bpfrpt_proptype_JobAlert === 'function' ? _JLCommon.bpfrpt_proptype_JobAlert : _propTypes2.default.shape(_JLCommon.bpfrpt_proptype_JobAlert)).apply(this, arguments);
    }).isRequired
};

// api/jobalert/me/

var bpfrpt_proptype_joblocationReverseLookupGetResult = {
    display_name: _propTypes2.default.string.isRequired
};

// api/joblocation-reverse-lookup

var bpfrpt_proptype_joblocationSuggestionGetResult = {
    suggestions: _propTypes2.default.arrayOf(_propTypes2.default.shape({
        value: _propTypes2.default.string.isRequired
    })).isRequired
};

// api/joblocation-suggestion

var bpfrpt_proptype_jobsearchSuggestionGetResult = {
    suggestions: _propTypes2.default.arrayOf(_propTypes2.default.shape({
        value: _propTypes2.default.string.isRequired,
        numJobs: _propTypes2.default.number
    })).isRequired
};

// api/jobsearch-suggestion

var bpfrpt_proptype_freqSearchSuggestionGetResult = {
    suggestions: _propTypes2.default.arrayOf(_propTypes2.default.shape({
        value: _propTypes2.default.string.isRequired,
        numJobs: _propTypes2.default.number.isRequired
    })).isRequired
};

// api/jobsearch-suggestion/popular

var bpfrpt_proptype_statisticalDataGetResult = {
    keys: _propTypes2.default.arrayOf(_propTypes2.default.string).isRequired,
    data: _propTypes2.default.object.isRequired
};

// api/charts

var bpfrpt_proptype_jobofferGetResult = _JLCommon.bpfrpt_proptype_Job === _propTypes2.default.any ? {} : _JLCommon.bpfrpt_proptype_Job;

// api/joboffer

var bpfrpt_proptype_popunderPostResponse = {
    showPopunder: _propTypes2.default.bool.isRequired,
    url: _propTypes2.default.string.isRequired,
    itemCount: _propTypes2.default.number.isRequired,
    partner: _propTypes2.default.string.isRequired,
    customer: _propTypes2.default.string.isRequired,
    customerId: _propTypes2.default.number.isRequired,
    suppressTransactionId: _propTypes2.default.bool,
    ch: _propTypes2.default.string.isRequired
};

// api/popunder

var bpfrpt_proptype_loginPostResult = {
    displayName: _propTypes2.default.string,
    firstName: _propTypes2.default.string.isRequired,
    initialCreated: _propTypes2.default.bool.isRequired,
    isAnonymous: _propTypes2.default.bool,
    lastName: _propTypes2.default.string.isRequired,
    localAccounts: _propTypes2.default.arrayOf(_propTypes2.default.shape({
        email: _propTypes2.default.string.isRequired
    })).isRequired,
    socialAccounts: _propTypes2.default.arrayOf(_propTypes2.default.object).isRequired,
    uniqueEmails: _propTypes2.default.arrayOf(_propTypes2.default.string).isRequired,
    uuid: _propTypes2.default.string.isRequired
};

// auth/json/login/local

var bpfrpt_proptype_directApplyDraftGet = _DirectApply.bpfrpt_proptype_DirectApplyDraft === _propTypes2.default.any ? {} : _DirectApply.bpfrpt_proptype_DirectApplyDraft;

// api/direct-apply/offer/<jobofferid>/draft

var bpfrpt_proptype_directApplyRecentFilesGet = _propTypes2.default.arrayOf(function () {
    return (typeof _DirectApply.bpfrpt_proptype_DirectApplyFile === 'function' ? _DirectApply.bpfrpt_proptype_DirectApplyFile : _propTypes2.default.shape(_DirectApply.bpfrpt_proptype_DirectApplyFile)).apply(this, arguments);
});

// api/direct-apply/files/recent


exports.bpfrpt_proptype_jpopunderPostResult = bpfrpt_proptype_jpopunderPostResult;
exports.bpfrpt_proptype_searchPostResult = bpfrpt_proptype_searchPostResult;
exports.bpfrpt_proptype_jobalertMeGetResult = bpfrpt_proptype_jobalertMeGetResult;
exports.bpfrpt_proptype_joblocationReverseLookupGetResult = bpfrpt_proptype_joblocationReverseLookupGetResult;
exports.bpfrpt_proptype_joblocationSuggestionGetResult = bpfrpt_proptype_joblocationSuggestionGetResult;
exports.bpfrpt_proptype_jobsearchSuggestionGetResult = bpfrpt_proptype_jobsearchSuggestionGetResult;
exports.bpfrpt_proptype_freqSearchSuggestionGetResult = bpfrpt_proptype_freqSearchSuggestionGetResult;
exports.bpfrpt_proptype_statisticalDataGetResult = bpfrpt_proptype_statisticalDataGetResult;
exports.bpfrpt_proptype_jobofferGetResult = bpfrpt_proptype_jobofferGetResult;
exports.bpfrpt_proptype_popunderPostResponse = bpfrpt_proptype_popunderPostResponse;
exports.bpfrpt_proptype_loginPostResult = bpfrpt_proptype_loginPostResult;
exports.bpfrpt_proptype_directApplyDraftGet = bpfrpt_proptype_directApplyDraftGet;
exports.bpfrpt_proptype_directApplyRecentFilesGet = bpfrpt_proptype_directApplyRecentFilesGet;"
`;
